app/src/main/java/com/markantoni/gofinandroid/MainActivity.java

```diff
@@ -83,7 +85,12 @@ public void showPattern(IPatternPresenter patternPresenter) {
     public void setPatternsList(List<IPatternPresenter> patterns) {
         mPagerAdapter = new PatternsPagerAdapter(getSupportFragmentManager(), patterns);
         mViewPager.setAdapter(mPagerAdapter);
-        showHint();
+        SharedPreferences sharedPreferences= PreferenceManager.getDefaultSharedPreferences(this);
+        boolean isFirstRun = sharedPreferences.getBoolean("firstrun", false);
```

Comment:
Could please extract `firstrun` as separate constant just to keep the code cleaner. Also I'd prefer to name it in way likely more `key.firstrun`